### PR TITLE
fix: color picker controlled value

### LINKED ISSUE #47539: ColorPicker当首次没有value和defaultValue时，通过其他方式手动赋值value，构件一直展示清除样式
### Reproduction link

[https://ant.design/components/color-picker-cn](https://ant.design/components/color-picker-cn)

### Steps to reproduce

```
import React, { useState } from "react";
import "./index.css";
import { ColorPicker, Space, Button } from "antd";

const Demo = () => {
  const [colorValue, setColorValue] = useState();
  return (
      <Space direction="vertical">
        <ColorPicker
          value={colorValue}
          onChange={setColorValue}
          showText
        />
        <Button
          onClick={() => {
            setColorValue("#FF0000");
          }}
        >
          Click
        </Button>
      </Space>
  );
};
export default Demo;
```

### What is expected?

按钮点击之后应该显示正常颜色，而不是清除的样式

### What is actually happening?

颜色未正常展示

| Environment | Info |
| --- | --- |
| antd | 5.14.1 |
| React | 18.0.0 |
| System | mac |
| Browser | 121.0.6167.184 |

### PR DESCRIPTION
**Pull request by MadCcc**
- fix: fix control value not show in dom
- chore: update title
- add test case
- change test title
- fix: cleared color should change after controlled value changed

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/47539

### 💡 Background and solution
在 `onChange` 返回的 Color 对象中添加 `isCleared` 字段，默认为 false，清楚颜色后变为 true。
这样做有一个缺陷是用户如果一直使用字符串颜色受控，如 hex string，那么清除颜色就不会展示出来。

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fix ColorPicker that displayed cleared color not change after `value` changed.         |
| 🇨🇳 Chinese |    修复 ColorPicker 展示的清除颜色在受控 `value` 变化后不会改变的问题。       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by lhojun** [comment]
this also have problem about clear with set color ""

**Review by li-jia-nan — DISMISSED** [review]

**Comment by yoyo837** [comment]
佬，这个还缺什么？希望能赶上本周的版本。

**Comment by afc163** [comment]
> this also have problem about clear with set color ""

@lhojun is right, this problem is still existed.

**Comment by MadCcc** [comment]
> I adjusted the test case and it does look like it hasn't been fully fixed yet.

我再看下

**Comment by yoyo837** [comment]
点错了

File: components/color-picker/__tests__/index.test.tsx [hunk 1]
```diff
@@ -607,4 +608,33 @@ describe('ColorPicker', () => {
...
+    expect(container.querySelector('.ant-color-picker-color-block-inner')).toHaveStyle({
+      background: 'rgb(255, 0, 0)',
+    });
+  });
+
+  it('When controlled and has an initial value, then changing the controlled value to cleared color should be reflected correctly on the DOM', async () => {
+    const Demo = () => {
+      const [color, setColor] = useState<ColorValueType>(generateColor('red'));
+      useEffect(() => {
+        // FIXME: How to cover '', null/undefined, generateColor('') at the same time?
```
Comment (yoyo837) [inline components/color-picker/__tests__/index.test.tsx:631]: 设置cleared颜色的方式有三种，用例怎么写可以都cover住？

Comment (MadCcc) [reply components/color-picker/__tests__/index.test.tsx:631]: '',null,undefined 都不代表 cleared 颜色，只有点击清除按钮之后的颜色是 cleared 颜色

Comment (MadCcc) [reply components/color-picker/__tests__/index.test.tsx:631]: 这里实际上是之前的逻辑对受控空值做了过滤

Comment (yoyo837) [reply components/color-picker/__tests__/index.test.tsx:631]: 意思是，受控设置'',null,undefined  无法从有颜色变为cleared色？

File: components/color-picker/__tests__/index.test.tsx [hunk 2]
```diff
@@ -607,4 +608,94 @@ describe('ColorPicker', () => {
     const { container } = render(<ColorPicker />);
     expect(container.querySelector('.ant-color-picker-clear')).toBeTruthy();
   });
+
+  ['', null].forEach((value) => {
```
Comment (yoyo837) [inline components/color-picker/__tests__/index.test.tsx:612]: 原来还可以这样玩...学会了。

**Review by yoyo837 — APPROVED** [review]

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**